### PR TITLE
Progress tabs

### PR DESCRIPTION
**Pull request by tmccoy529**
Added the css to handle the look of progress tabs. Added an actual generic component as well to display inside of the library.

### CODE REVIEW

File: src/components/foundation/Grid.md [hunk 1]
```diff
@@ -52,11 +52,15 @@ another use please submit a
...
+  <IconButton text="Fish Button" icon="fas fa-fish" description="" />
+  <IconButton text="Fish Button" icon="fas fa-fish" description="" />
+  <IconButton text="Fish Button" icon="fas fa-fish" description="" />
+  <IconButton text="Fish Button" icon="fas fa-fish" description="" />
+  <IconButton text="Fish Button" icon="fas fa-fish" description="" />
+  <IconButton
+    text="Really Long, Long, Fish Button"
+    icon="fas fa-fish"
+    description=""
+  />
```
Comment (tmccoy529) [inline src/components/foundation/Grid.md:84]: added a null description to fix a console error but can be backed out if need be

File: src/components/basic/ProgressTabs.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import classnames from "classnames";
+
+const ProgressTabs = (props) => {
+  const { className, tabItems = [] } = props;
+  const cssClasses = classnames("dg_progress-tabs", className);
+
+  return (
+    <div>
+      <ol className={cssClasses}>
+        {tabItems.map((item) => {
```
Comment (martypowell) [inline src/components/basic/ProgressTabs.js:13]: So in general from what i'm seeing the react community they recommend taking this mapped item and making it it's own component so can we create `ProgressTabsItem` the li?

File: src/dotgov.scss [hunk 3]
```diff
@@ -81,6 +81,7 @@
 @import "./styles/partials/components/content";
 @import "./styles/partials/components/explore";
 @import "./styles/partials/components/scrollable-container";
+@import "./styles/partials/components/progresstabs";
```
Comment (martypowell) [inline src/dotgov.scss]: can you change the name of this file to `progress-tabs` to match the class name?

File: src/styles/partials/components/_progresstabs.scss [hunk 4]
```diff
@@ -0,0 +1,80 @@
+.dg_progress-tabs {
```
Comment (martypowell) [inline src/styles/partials/components/_progresstabs.scss]: this looking pretty good but is missing the media queries. you should be able to grab those from the baltcogo-reporter css

File: src/styles/partials/components/_progresstabs.scss [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  display: flex;
+  list-style-type: none;
+  padding: 0;
+
+  li {
+    background-color: $gray-lightest;
+    color: $gray;
+    flex: 1 1 auto;
+    font-family: "Montserrat", sans-serif;
```
Comment (martypowell) [inline src/styles/partials/components/_progresstabs.scss]: this should use a variable, not sure what it is off the top of my head.

File: src/components/foundation/Grid.md (hunk 1 above)
Comment (martypowell) [reply src/components/foundation/Grid.md:84]: This is good, but this is actually a required field per https://baltimorecounty.github.io/dotgov-components/#iconbutton

Can you please just add "Visit our fishes dashboard to learn more." to each description?

**Review by martypowell — CHANGES_REQUESTED** [review]
most important is the media query stuff.

File: src/styles/partials/components/_progresstabs.scss (hunk 4 above)
Comment (tmccoy529) [reply src/styles/partials/components/_progresstabs.scss]: yup I forgot about those. Will add them now

File: src/components/foundation/Grid.md (hunk 1 above)
Comment (tmccoy529) [reply src/components/foundation/Grid.md:84]: done

File: src/dotgov.scss (hunk 3 above)
Comment (tmccoy529) [reply src/dotgov.scss]: done

File: src/styles/partials/components/_progresstabs.scss (hunk 4 above)
Comment (tmccoy529) [reply src/styles/partials/components/_progresstabs.scss]: added

File: src/styles/partials/components/_progresstabs.scss (hunk 5 above)
Comment (tmccoy529) [reply src/styles/partials/components/_progresstabs.scss]: hahaha yeah its the default font. Thanks for the catch.

File: src/components/basic/ProgressTabs.js (hunk 2 above)
Comment (tmccoy529) [reply src/components/basic/ProgressTabs.js:13]: broke it out

File: src/styles/partials/components/_progress-tabs.scss [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+      &:after {
+        border: none;
+      }
+    }
+  }
+}
+
+@media (max-width: $medium-breakpoint) {
+  .dg_progress-tabs {
```
Comment (martypowell) [inline src/styles/partials/components/_progress-tabs.scss:84]: i am still getting some weird display on smaller screens.

image

Comment (martypowell) [reply src/styles/partials/components/_progress-tabs.scss:84]: I think maybe the issues is...

&:after,
&:before,
&:last-child

Not tested though

**Review by martypowell — CHANGES_REQUESTED** [review]
minor media query stuff

File: src/styles/partials/components/_progress-tabs.scss (hunk 6 above)
Comment (tmccoy529) [reply src/styles/partials/components/_progress-tabs.scss:84]: resolved

File: src/styles/partials/components/_progress-tabs.scss [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    display: block;
+    padding-right: $padding-large;
```
Comment (martypowell) [inline src/styles/partials/components/_progress-tabs.scss]: this can be removed.

File: src/styles/partials/components/_progress-tabs.scss [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    list-style-type: decimal;
+    list-style-position: inside;
+    li {
+      display: list-item;
```
Comment (martypowell) [inline src/styles/partials/components/_progress-tabs.scss]: the x'd out fields can be removed

image

File: src/styles/partials/components/_progress-tabs.scss (hunk 7 above)
Comment (tmccoy529) [reply src/styles/partials/components/_progress-tabs.scss]: done

File: src/styles/partials/components/_progress-tabs.scss (hunk 8 above)
Comment (tmccoy529) [reply src/styles/partials/components/_progress-tabs.scss]: done

**Review by martypowell — APPROVED** [review]

**Pull request merged**